hazelcast/src/main/java/com/hazelcast/config/security/LdapAuthenticationConfig.java

```diff
@@ -0,0 +1,406 @@
+            return false;
+        }
+        if (roleFilter == null) {
+            if (other.roleFilter != null) {
+                return false;
+            }
+        } else if (!roleFilter.equals(other.roleFilter)) {
+            return false;
+        }
+        if (roleMappingAttribute == null) {
+            if (other.roleMappingAttribute != null) {
+                return false;
+            }
+        } else if (!roleMappingAttribute.equals(other.roleMappingAttribute)) {
+            return false;
+        }
+        if (roleMappingMode != other.roleMappingMode) {
+            return false;
+        }
+        if (roleNameAttribute == null) {
+            if (other.roleNameAttribute != null) {
+                return false;
+            }
+        } else if (!roleNameAttribute.equals(other.roleNameAttribute)) {
+            return false;
+        }
+        if (roleRecursionMaxDepth != other.roleRecursionMaxDepth) {
+            return false;
+        }
+        if (roleSearchScope != other.roleSearchScope) {
+            return false;
+        }
+        if (socketFactoryClassName == null) {
+            if (other.socketFactoryClassName != null) {
+                return false;
+            }
+        } else if (!socketFactoryClassName.equals(other.socketFactoryClassName)) {
+            return false;
+        }
+        if (systemUserDn == null) {
+            if (other.systemUserDn != null) {
+                return false;
+            }
+        } else if (!systemUserDn.equals(other.systemUserDn)) {
+            return false;
+        }
+        if (systemUserPassword == null) {
+            if (other.systemUserPassword != null) {
+                return false;
+            }
+        } else if (!systemUserPassword.equals(other.systemUserPassword)) {
+            return false;
+        }
+        if (url == null) {
+            if (other.url != null) {
+                return false;
+            }
+        } else if (!url.equals(other.url)) {
+            return false;
+        }
+        if (userContext == null) {
+            if (other.userContext != null) {
+                return false;
+            }
+        } else if (!userContext.equals(other.userContext)) {
+            return false;
+        }
+        if (userFilter == null) {
+            if (other.userFilter != null) {
+                return false;
+            }
+        } else if (!userFilter.equals(other.userFilter)) {
+            return false;
+        }
+        if (userNameAttribute == null) {
+            if (other.userNameAttribute != null) {
+                return false;
+            }
+        } else if (!userNameAttribute.equals(other.userNameAttribute)) {
+            return false;
+        }
+        if (userSearchScope != other.userSearchScope) {
+            return false;
+        }
+        return true;
+    }
+
+    private void setIfConfigured(Properties props, String propertyName, String value) {
+        if (!isNullOrEmpty(value)) {
```

Comment:
Is there a real scenario when the value is `null`?

Reply:
Yes, most of the properties in the login module is optional.